Library/include/CSP/Systems/HotSpotSequence/HotSpotSequenceSystem.h

```diff
@@ -0,0 +1,105 @@
+
+} // namespace csp::memory
+
+namespace csp::systems
+{
+class SequenceSystem;
+class SpaceSystem;
+/// @ingroup HotSpotSequenceSystem System
+/// @brief Public facing system that allows the management of groupings of hotspots in a space.
+
+class CSP_API HotSpotSequenceSystem : public SystemBase
+{
+public:
+	CSP_START_IGNORE
+	/** @cond DO_NOT_DOCUMENT */
+	friend class SystemsManager;
+	friend void csp::memory::Delete<HotSpotSequenceSystem>(HotSpotSequenceSystem* Ptr);
+	/** @endcond */
+	CSP_END_IGNORE
+
+	HotSpotSequenceSystem(csp::systems::SequenceSystem* SequenceSystem, csp::systems::SpaceSystem* SpaceSystem);
+	/// @brief Create a Hotspot group
+	/// @param GroupName csp::common::String : The unique grouping name
+	/// @param HotspotIds csp::common::Array<csp::common::String> : set of Hotspot ids to add to the group
+	/// @param Callback HotSpotGroupResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void CreateHotspotGroup(const csp::common::String&,
+											 const csp::common::Array<csp::common::String>& HotspotIds,
+											 HotSpotGroupResultCallback Callback);
+
+	/// @brief Rename a Hotspot group
+	/// @param CurrentGroupName csp::common::String : The unique grouping name
+	/// @param NewGroupName csp::common::String : The unique grouping name
+	/// @param Callback HotSpotGroupResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void
+		RenameHotspotGroup(const csp::common::String& GroupName, const csp::common::String& NewGroupName, HotSpotGroupResultCallback Callback);
+	/// @brief Update a Hotspot group
+	/// @param GroupName csp::common::String : The unique grouping name
+	/// @param HotspotIds csp::common::Array<csp::common::String> : set of Hotspot ids to add to the group
+	/// @param Callback HotSpotGroupResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void UpdateHotspotGroup(const csp::common::String& GroupName,
+											 const csp::common::Array<csp::common::String>& HotspotIds,
+											 HotSpotGroupResultCallback Callback);
+
+	/// @brief Get a Hotspot group by name
+	/// @param GroupName csp::common::String : The unique grouping name
+	/// @param Callback HotSpotGroupResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void GetHotspotGroup(const csp::common::String& GroupName, HotSpotGroupResultCallback Callback);
+
+
+	/// @brief Get all Hotspot groups for the current space and logged in user.
+	/// @param Callback HotSpotGroupsResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void GetHotspotGroups(HotSpotGroupsResultCallback Callback);
+
+
+	/// @brief Get a Hotspot group by name
+	/// @param GroupName csp::common::String : The unique grouping name
+	/// @param Callback HotSpotGroupResultCallback : callback to call when a response is received
+	CSP_ASYNC_RESULT void DeleteHotspotGroup(const csp::common::String& GroupName, HotSpotGroupResultCallback Callback);
```

Comment:
Is this the right callback? The response would be success or failure here I would think? No need to return a hotspot group.